gc/base/standard/Scavenger.cpp

```diff
@@ -1707,6 +1707,36 @@ MM_Scavenger::scavengeObjectSlots(MM_EnvironmentStandard *env, MM_CopyScanCacheS
 	return shouldRemember;
 }
 
+void
+MM_Scavenger::deepScanOutline(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr, uintptr_t selfReferencingField1, uintptr_t selfReferencingField2){
+
+	void* tempObj = objectPtr;
+	uintptr_t priorityField = selfReferencingField1;
+
+	//Throttle - Deep scan should be terminated when the free list is utilized more than 50%
+	uintptr_t freeListUtilizationLimit = _scavengeCacheFreeList.getAllocatedCacheCount() / 2;
+
+	while(true) {
+		GC_SlotObject tempSlot(env->getOmrVM(), (fomrobject_t*)(((uintptr_t)tempObj)+priorityField));
+
+		if(tempSlot.readReferenceFromSlot() == NULL){
+
+			if(priorityField == selfReferencingField2 || selfReferencingField2 == 0){
+				break;
+			}
+
+			priorityField = selfReferencingField2;
+		} else{
+			copyAndForward(env, &tempSlot);
+
```

Comment:
put a comment: did we encounter an already visited object or hit throttling threshold?